LibOS/shim/src/shim_malloc.c

```diff
@@ -360,20 +362,27 @@ void * calloc (size_t nmemb, size_t size)
 }
 extern_alias(calloc);
 
-void* realloc(void* ptr, size_t new_size)
+#if 0 /* Temporarily disabling this code */
+void * realloc(void * ptr, size_t new_size)
 {
-    // TODO: This function should be optimized to avoid memcpy when buffer
-    // expansion is possible.
+    /* TODO: We can't deal with this case right now */
+    assert(!MEMORY_MIGRATED(ptr));
+
     size_t old_size = slab_get_buf_size(slab_mgr, ptr);
-    void* new_buf = malloc(new_size);
+    if (old_size >= new_size)
+        return ptr;
```

Comment:
I'm not sure if this if is a good idea, at least without any thresholds. The caller may expect that if they realloc a 10MB array into a single byte then the memory consumption will be low.

Reply:
Technically, this is GLIBC behavior. However, I agree with your assessment of the potential risk of memory leaking. I added a comment to make sure we deal with the case when resurrecting this code.

Reply:
Looks better now, thanks for adding the comment.